modules/libcom/src/osi/osiSock.h

```diff
@@ -28,185 +34,367 @@ struct sockaddr;
  */
 LIBCOM_API unsigned epicsStdCall ipAddrToHostName (
     const struct in_addr * pAddr, char * pBuf, unsigned bufSize );
 
-/*
- * attempt to convert ASCII string to an IP address in this order
- * 1) look for traditional doted ip with optional port
- * 2) look for raw number form of ip address with optional port
- * 3) look for valid host name with optional port
+/*!
+ * \brief Attempt to convert ASCII string to an IP address
+ *
+ * Attempt to convert ASCII string to an IP address. Conversion occurs this
+ * order:
+ * 1. look for traditional doted ip with optional port
+ * 2. look for raw number form of ip address with optional port
+ * 3. look for valid host name with optional port
+ *
+ * \param pAddrString IP address string to convert to a sockaddr_in (e.g. "192.168.0.1:80")
+ * \param defaultPort The default port (service) to use if not specified in pAddrString
+ * \param[out] pIP Pointer to a sockaddr_in where the converted address info will be stored
+ * \return negative value on error, 0 on success.
  */
 LIBCOM_API int epicsStdCall aToIPAddr
     ( const char * pAddrString, unsigned short defaultPort, struct sockaddr_in * pIP);
 
-/*
- * attempt to convert ASCII host name string with optional port to an IP address
+/*!
+ * \brief Attempt to convert ASCII host name string with optional port to an IP address
+ *
+ * Convert an in_addr struct to a human readable hostname.
+ *
+ * \param pHostName ASCII host name to convert
+ * \param[out] pIPA Pointer to the in_addr structure where the hostname information will be stored
+ * \return 0 on success, negative on error.
  */
 LIBCOM_API int epicsStdCall hostToIPAddr 
     (const char *pHostName, struct in_addr *pIPA);
-/*
- * attach to BSD socket library
+/*!
+ * \brief attach to BSD socket library
+
+ * Attach to the BSD socket library in preparation for using sockets.  If the OS
+ * does not require attaching to the socket library, this function does nothing.
+ *
+ * \return true on success, false on error.
  */
 LIBCOM_API int epicsStdCall osiSockAttach (void); /* returns T if success, else F */
 
-/*
- * release BSD socket library
+/*!
+ * \brief release BSD socket library
+ *
+ * Release the BSD socket library.if the OS requires it.  For platforms that do
+ * not require attaching and releasing to use the socket library, this function
+ * does nothing.
  */
 LIBCOM_API void epicsStdCall osiSockRelease (void);
 
-/*
- * convert socket error numbers to a string
+/*!
+ * \brief convert socket error numbers to a string
+ *
+ * There are several system functions which set errno on failure.  This function
+ * converts that error number into a string describing the error.
+ *
+ * \param[out] pBuf Pointer to char array where the error string will be stored
+ * \param bufSize Length of the array pointed to by pBuf
+ * \param error error number to convert to a string
+ * will convert such an errno to a human readable string.
```

Comment:
Is this intended to be a single sentence for the `error` parameter?

Reply:
Not sure what happened there.  It now reads:
 
 * \param error The error number to describe in string form